include/swift/Driver/ToolChain.h

```diff
@@ -13,6 +13,7 @@
 #ifndef SWIFT_DRIVER_TOOLCHAIN_H
 #define SWIFT_DRIVER_TOOLCHAIN_H
 
+#include "swift/AST/DiagnosticEngine.h"
```

Comment:
You can avoid this include (here and in DarwinToolChain.h) by forward-declaring DiagnosticEngine in the `swift` namespace. That helps (a tiny bit) with incremental rebuilds.